components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -193,10 +193,13 @@ public static AuthenticatedUser getAuthenticatedUser(String fullyQualifiedUserNa
      * IdentityOAuthAdminException exception
      * @param message error message
      * @param exception Exception.
+     * @param isErrorLogged decide whether the message should be optionally logged as an error
      * @return
      */
-    public static IdentityOAuthAdminException handleError(String message, Exception exception) {
-        log.error(message);
+    public static IdentityOAuthAdminException handleError(String message, Exception exception, Boolean isErrorLogged) {
```

Comment:
Since this is a public method, changing the method signature will break the backward compatibility. Better to introduce a new method for this requirement.